Assets/Scripts/Data/DataModels/Missions/Mission.cs

```diff
@@ -44,6 +53,29 @@ public async Task LoadDataAsync()
         saveData = await DataUtils.LoadFileAsync<MissionSaveData>(name, FOLDER_NAME);
     }
 
+    public void UnlockIfConditionMet()
+    {
+        switch (UnlockCondition)
+        {
+            case MissionUnlockCondition.TotalMoney:
+                // This is called back by the Player Manager's OnFinancialTransaction() event. 
+                HasBeenUnlocked = CanBeUnlockedWithMoney;
+
+                if (HasBeenUnlocked)
```

Comment:
I don't really like nesting an if in a switch, but can't see a better way, and I don't think it's outrageous to do it here. 

Could make a separate private method that sets the flag and handles the event subscription, but that would basically be doing the same thing just abstracted away.